core/contract.go

```diff
@@ -64,3 +64,18 @@ type Contract struct {
 	// Root of the contract's storage trie.
 	StorageRoot *felt.Felt // TODO: is this field necessary?
 }
+
+// ContractAddress computes the address of a StarkNet contract.
+func ContractAddress(callerAddress, classHash, salt *felt.Felt, constructorCalldata []*felt.Felt) *felt.Felt {
```

Comment:
Drive-by comment: intuitively, I feel this should be a method on `Contract` so we can do something like `contract.Address()`. It feels weird to me that we need information from outside the `Contract` to calculate the address, especially since a contract is stored [by its address](https://docs.starknet.io/documentation/architecture_and_concepts/State/starknet-state/) in the trie.

Happy to hear different opinions on this.

Reply:
I have to agree. However, for now I will add it as a todo, rebase and merge the branch.